psycopg/psycopg/types/geometry.py

```diff
@@ -0,0 +1,63 @@
+"""
+Adapters for PostGIS geometries
+"""
+
+from typing import TYPE_CHECKING
+
+from psycopg.adapt import Dumper, Loader
+from psycopg.pq import Format
+from psycopg.types import TypeInfo
+
+if TYPE_CHECKING:
+    from psycopg.connection import Connection
+    from psycopg.rows import Row
+
+
+def register_shapely_adapters(conn: Connection[Row]) -> None:
+    """Register Shapely dumper and loaders.
+
+    After invoking this function on a connection, the queries retrieving
+    PostGIS geometry objects will return Shapely's shape object instances
+    in text and binary mode.
+
+    Similarly, shape onjects can be sent to the database.
+
+    :param conn: the connection on which to set up the loader and dumper
+
+    """
+    import shapely.wkb
```

Comment:
Better warn the user if they need an extra package: see
https://github.com/psycopg/psycopg/blob/2c991429cc28ca50b5ad292f5e42ad5a85b4cf46/psycopg/psycopg/_dns.py#L21-L24
for example.

I wonder if we shouldn't add an extra, so that people may install `psycopg[shapely]` to get a shapely of the right version bracket. Or is this best left to the user?

Reply:
I think it's best left to the user, installing it requires GEOS and usually is done with extra libraries to handle projections. Also there are different versions actively used, 1.7, the beta for 1.8 and the future 2.0 (I tested using 1.7 and 1.8 beta)